chain/dependencies.go

```diff
@@ -197,10 +197,14 @@ type Object interface {
 	//
 	// -1 means no start/end
 	ValidRange(Rules) (start int64, end int64)
+}
 
+type HasMarshal interface {
 	// Marshal encodes an [Action] as bytes.
 	Marshal(p *codec.Packer)
+}
 
+type HasSize interface {
```

Comment:
Could we define this as a single `Marshaller` interface? The recommended pattern in golang is to name single function interfaces as "action"-er, so `Marshaller` would be appropriate in this case. I think it makes sense to use just Marshaller because there's no reason to implement one and not the other.

Someone that defines their own `Marshal` function should also define their own size function instead of using the linear codec's size function. Falling back to it won't error, but it doesn't make sense to implement one and not the other.